src/device-manager/java/src/nl/Weave/DataManagement/WdmClient.java

```diff
@@ -66,12 +74,15 @@
      * that has been assigned via {@link #setCompletionHandler}.
      */
     public void beginRefreshData();
+    public void beginRefreshDataWithEvents();
 
     public CompletionHandler getCompletionHandler();
     public void setCompletionHandler(CompletionHandler compHandler);
 
     public GenericTraitUpdatableDataSink getDataSink(long traitInstancePtr);
 
+    public String getEvents();
```

Comment:
What is the format of this string? Shouldn't this be a List<String> or at least String[] array?

Reply:
The return value will be a string in json format

Reply:
Got it. Can you add that as Javadoc, and also indicate that the beginFetchEvents() must complete successfully first? Would be good to document what it is expected to return if events have not been fetched, too. An empty JSON object? Or an empty string, or null?